### PR TITLE
fix(ai): llm model moved to share state to guarantee consistency

### PR DESCRIPTION
**Pull request by tysoekong**
### Summary

Sets request model name consistently across AI plugins.

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [x] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Review by fffonion — APPROVED** [review]

**Review by oowl — APPROVED** [review]

**Comment by tysoekong** [comment]
@fffonion Added integration test now for this, please verify once more

**Review by fffonion — APPROVED** [review]
wait for CI

**Comment by tysoekong** [comment]
@fffonion Should pass now

File: kong/plugins/ai-response-transformer/handler.lua [hunk 1]
```diff
@@ -8,6 +8,7 @@ local kong_utils    = require("kong.tools.gzip")
 local llm           = require("kong.llm")
 local llm_state     = require("kong.llm.state")
 local ai_shared     = require("kong.llm.drivers.shared")
+local llm_state     = require("kong.llm.state")
```
Comment (fffonion) [inline kong/plugins/ai-response-transformer/handler.lua:11]:
```suggestion
```

File: kong/plugins/ai-request-transformer/handler.lua [hunk 2]
```diff
@@ -6,6 +6,7 @@ local fmt           = string.format
 local llm           = require("kong.llm")
 local llm_state     = require("kong.llm.state")
 local ai_shared     = require("kong.llm.drivers.shared")
+local llm_state     = require("kong.llm.state")
```
Comment (fffonion) [inline kong/plugins/ai-request-transformer/handler.lua:9]:
```suggestion
```

**Comment by fffonion** [comment]
two lint errors, rerunning CI

**Comment by fffonion** [comment]
https://github.com/Kong/kong-ee/pull/10206
https://github.com/Kong/kong-ee/pull/10207
https://github.com/Kong/kong-ee/pull/10208

**Pull request merged**